src/main/java/com/endava/web/StudentController.java

```diff
@@ -0,0 +1,64 @@
+package com.endava.web;
+
+import com.endava.repository.Student;
+import com.endava.service.StudentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/student")
+public class StudentController {
+
+    private StudentService studentService;
+
+    @Autowired
+    public StudentController(StudentService studentService) {
+        this.studentService = studentService;
+    }
+
+    @RequestMapping(value = "/id", method = RequestMethod.GET, produces = "application/json")
```

Comment:
why "/id" ?

Reply:
Originally it was /student, and when new endpoint is added and both of them use GET on the same path, so I changed both of them (at least one of them should be changed):
-  /student/id is for finding student when id is given 
- /student/name is for finding students when name is given

I will reverse it, and for new endpoint leave student/name.